propagator/vitess/test/opentelemetry/propagator/vitess/sql_query_propagator_test.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::Propagator::Vitess::SqlQueryPropagator do
+  let(:span_id) do
+    'e457b5a2e4d86bd1'
+  end
+
+  let(:trace_id) do
+    '80f198ee56343ba864fe8b2a57d3eff7'
+  end
+
+  let(:trace_flags) do
+    OpenTelemetry::Trace::TraceFlags::DEFAULT
+  end
+
+  let(:context) do
+    OpenTelemetry::Trace.context_with_span(
+      OpenTelemetry::Trace.non_recording_span(
+        OpenTelemetry::Trace::SpanContext.new(
+          trace_id: Array(trace_id).pack('H*'),
+          span_id: Array(span_id).pack('H*'),
+          trace_flags: trace_flags
+        )
+      )
+    )
+  end
+
+  let(:propagator) do
+    OpenTelemetry::Propagator::Vitess::SqlQueryPropagator.new
+  end
+
+  let(:parent_context) do
+    OpenTelemetry::Context.empty
+  end
+
+  let(:base64_encoded_jaeger_context) do
+    Base64.strict_encode64({'uber-trace-id' => "#{trace_id}:#{span_id}:0:#{trace_flags.sampled? ? 1 : 0}"}.to_json)
+  end
+
+  let(:carrier) do
+    +'SELECT * FROM users'
```

Comment:
the `+` here is to deal with the frozen string issue, right?

Reply:
Yep